### DISCUSSION TITLE
Issue when running run_cgs_models_mallet() prior to scenic+

### DISCUSSION THREAD

**Discussion by laurajohanne** [aertslab/pycisTopic#122] [General]
Hi,

First I once again have to emphasize would a much needed and organized packages you have made! And enjoy working with it.

I have been running the scenic+ pipeline on a non-multiome mouse liver datasets.

I have succesfully used both the previous pipeline and afterwards the snake make pipeline (and was really happy for the optimized and much faster pipeline!) with no issues.

I then wanted to proceed on new datasets but I am having some issues when running  run_cgs_models_mallet().

Code;
models=run_cgs_models_mallet(path_to_mallet_binary,
                    cistopic_obj,
                    n_topics=[2,5,10,15,20,25,30,35,40,45,50],
                    n_cpu=55,
                    n_iter=500,
                    random_state=1,
                    alpha=50,
                    alpha_by_topic=True,
                    eta=0.1,
                    eta_by_topic=False,
                    tmp_path= tmpDir, #Use SCRATCH if many models or big data set
                    save_path = None)

The error;
TypeError: run_cgs_models_mallet() got multiple values for argument 'n_topics'

Version of pycisTopic:
'2.0a0'

When looking at the source code n_topics:
n_topics: list of int
        A list containing the number of topics to use in each model.

Which mine is; n_topics=[2,5,10,15,20,25,30,35,40,45,50].

So I am not sure how to proceed. Hope you can help.

Thanks in advance!

Best regards, 
Laura

**Comment by SeppeDeWinter** [comment]
Hi @laurajohanne 

Thank you for the nice comments.

The error is thrown because you did not provide an argument name for the second argument (which is n_topics, see https://github.com/aertslab/pycisTopic/blob/e5d5f19c941c7416628a80c846848eb49c3167b9/src/pycisTopic/lda_models.py#L734).

Can you try again with the following code:

```python

models=run_cgs_models_mallet(
  mallet_path = path_to_mallet_binary,
  cistopic_obj = cistopic_obj,
  n_topics=[2,5,10,15,20,25,30,35,40,45,50],
  n_cpu=55,
  n_iter=500,
  random_state=1,
  alpha=50,
  alpha_by_topic=True,
  eta=0.1,
  eta_by_topic=False,
  tmp_path= tmpDir, #Use SCRATCH if many models or big data set
  save_path = None)

```

All the best,

Seppe

**Reply by laurajohanne** [reply]
Thank you very much that worked!

**Reply by SeppeDeWinter** [reply]
Great

Happy to help.

Seppe
